client/sql/dataframe.go

```diff
@@ -31,6 +31,7 @@ type DataFrame interface {
 	Show(numRows int, truncate bool) error
 	Schema() (*StructType, error)
 	Collect() ([]Row, error)
+	Write() DataFrameWriter
 }
 
 type dataFrameImpl struct {
```

Comment:
I'm wondering if we should split the interfaces and the impls across multiple files for readability?

Reply:
How about split the file when it becomes big (e.g. more than 1000 or 2000 lines)?

In Golang, it is difficult to find out which structs implement which interfaces, since Go structs does not explicitly declare what interfaces they implement (Go uses method matching to determine what interfaces a struct implements). If a file is not huge, placing interface and implementation struct in the same file will make it easy to track the implementation relationship of interfaces/structs.

Reply:
Sounds good. I realized the problem of the open interfaces as well :(